packages/@sanity/base/src/components/formField/FormFieldSet.tsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable camelcase */
+
+import {Path, Marker} from '@sanity/types'
+import {Box, Flex, Grid, Stack, Text, useForwardedRef} from '@sanity/ui'
+import {FOCUS_TERMINATOR} from '@sanity/util/paths'
+import React, {forwardRef, useState, useCallback, useEffect} from 'react'
+import styled, {css} from 'styled-components'
+import {ChangeIndicator, ChangeIndicatorContextProvidedProps} from '../../change-indicators'
+import {FieldPresence, FormFieldPresence} from '../../presence'
+import {FormFieldValidationStatus} from './FormFieldValidationStatus'
+import {FormFieldSetLegend} from './FormFieldSetLegend'
+import {markersToValidationList} from './helpers'
+
+export interface FormFieldSetProps {
+  /**
+   * @beta
+   */
+  __unstable_changeIndicator?: ChangeIndicatorContextProvidedProps | boolean
+  /**
+   * @beta
+   */
+  __unstable_markers?: Marker[]
+  /**
+   * @beta
+   */
+  __unstable_presence?: FormFieldPresence[]
+  children: React.ReactNode
+  collapsed?: boolean
+  collapsible?: boolean
+  columns?: number
+  description?: React.ReactNode
+  /**
+   * The nesting level of the form field set
+   */
+  level?: number
+  onFocus?: (path: Path) => void
+  onToggle?: (collapsed: boolean) => void
+  title?: React.ReactNode
+}
+
+const FOCUS_PATH = [FOCUS_TERMINATOR]
+
+const Root = styled(Box).attrs({forwardedAs: 'fieldset'})`
+  border: none;
+
+  /* See: https://thatemil.com/blog/2015/01/03/reset-your-fieldset/ */
+  body:not(:-moz-handler-blocked) & {
+    display: table-cell;
+  }
+`
+
+const Content = styled(Box)<{$borderLeft: boolean}>(({$borderLeft}) => {
```

Comment:
Why the `$` prefix here? Perhaps add a comment about what it's for?

Reply:
It's the convention for "transient props" in `styled-components`: https://styled-components.com/docs/api#transient-props

Will add a comment about it.

Reply:
Ah, that's a neat convention.